XTS/WS-C/dev/src/com/arjuna/webservices11/wscoor/client/WSCOORClient.java

```diff
@@ -41,12 +42,27 @@
     private static synchronized ActivationService getActivationService()
     {
         if (activationService.get() == null) {
-            // activationService.set(new ActivationService(null, new QName("http://docs.oasis-open.org/ws-tx/wscoor/2006/06", "ActivationService")));
-            activationService.set(new ActivationService());
+            setActivationService();
         }
         return activationService.get();
     }
 
+    private static synchronized void setActivationService()
+    {
+        // activationService.set(new ActivationService(null, new QName("http://docs.oasis-open.org/ws-tx/wscoor/2006/06", "ActivationService")));
+        if (System.getSecurityManager() == null) {
+            activationService.set(new ActivationService());
+        } else {
+            AccessController.doPrivileged(new PrivilegedAction<Void>() {
+                @Override
+                public Void run() {
+                    activationService.set(new ActivationService());
```

Comment:
All these services are extensions of javax.xml.ws.Service class. They don't do much other than call super methods (this particular constructor also only calls super). But since they're always created from the client, I thought it'd be better to check permissions in the client rather than scatter them through the bunch of different classes.